presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ActualProperties.java

```diff
@@ -372,14 +372,14 @@ private boolean isCoordinatorOnly()
             return nodePartitioning.get().getHandle().isCoordinatorOnly();
         }
 
-        private boolean isNodePartitionedOn(Collection<Symbol> columns, Set<Symbol> constants, boolean nullsReplicated)
+        private boolean isNodePartitionedOn(Collection<Symbol> columns, Set<Symbol> constants, Replication replication)
         {
-            return nodePartitioning.isPresent() && nodePartitioning.get().isPartitionedOn(columns, constants) && this.nullsReplicated == nullsReplicated;
+            return nodePartitioning.isPresent() && nodePartitioning.get().isPartitionedOn(columns, constants) && Objects.equals(this.replication, replication);
```

Comment:
I prefer `Objects.equals(this.replication, replication)` to `this.replication.equals(replication)`